plugin/trino-hive/src/main/java/io/trino/plugin/hive/avro/AvroFileWriterFactory.java

```diff
@@ -117,7 +117,7 @@ public Optional<FileWriter> createFileWriter(
             Closeable rollbackAction = () -> fileSystem.deleteFile(location);
 
             return Optional.of(new AvroHiveFileWriter(
-                    outputFile.create(outputStreamMemoryContext),
+                    outputFile.write(outputStreamMemoryContext),
```

Comment:
Trying to get caught up on context here. Is the idea that Avro first needs to first write to memory before flushing? If so, is this unique to Avro or just this didn't follow the right pattern?

Reply:
And it's not unique to Avro, that's just the example usage that Piotr provided